### PR TITLE
Implement lazy area weights

### PR DESCRIPTION
**Pull request by schlunma**
### Description

This PR makes `iris.analysis.cartography.area_weights` lazy by providing the keyword arguments `compute` and `chunks`:

```py
def area_weights(cube, normalize=False, compute=True, chunks=None):
```

These defaults ensure full backwards-compatibility. ~`chunks=None` results in using the cube data's chunks for the weights array, which is a resonable default since these weights will probably be used with the cube's data anyway.~

~This requires https://github.com/SciTools/iris/pull/5620, so still in draft mode. Apart from that, this PR is ready.~

Closes https://github.com/SciTools/iris/issues/5611

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by trexfeathers** [comment]
Thanks for the hard work @schlunma. #5620 will be reviewed by @pp-mo in the new year, so #5658 can get moving again once that is sorted.

File: lib/iris/tests/test_analysis.py [hunk 1]
```diff
@@ -1155,6 +1155,55 @@ def test_area_weights(self):
...
+        # Check that chunksizes are as expected
+        if chunks is not None:
+            assert area_weights.chunksize == chunks
+        else:
+            if isinstance(cube_data, da.Array):
+                assert area_weights.chunksize == cube_data.chunksize
+            else:
+                assert area_weights.chunksize == (4, 3, 4)
+
+        # Check that actual weights are as expected
```
Comment (trexfeathers) [inline lib/iris/tests/test_analysis.py:1190]: Please could you note that these are 'known good output'? I have previously panicked when dependencies have changed test outputs and I had no idea why the original numbers were there!

File: lib/iris/tests/test_analysis.py [hunk 2]
```diff
@@ -1155,6 +1155,55 @@ def test_area_weights(self):
         )
 
 
+@tests.skip_data
+class TestLazyAreaWeights:
```
Comment (trexfeathers) [inline lib/iris/tests/test_analysis.py:1159]: I'm happy with the coverage from this 👍

**Review by trexfeathers — CHANGES_REQUESTED** [review]
This looks great, thanks @schlunma! Just a tiny request please

File: lib/iris/tests/test_analysis.py (hunk 1 above)
Comment (schlunma) [reply lib/iris/tests/test_analysis.py]: Done in https://github.com/SciTools/iris/pull/5658/commits/8a03706d678de744f616a94fd6a0f87b048c97c1, is this okay like this or do I need to add more?

**Comment by schlunma** [comment]
Just had a second look on this. I think it's actually better to use `cube.lazy_data().chunks` instead of `cube.lazy_data().chunksize` to make sure that the chunk structure is exactly identical in the weights and the original data. It probably won't matter, but it feels better to use `.chunks` here...

**Comment by schlunma** [comment]
After having a third look on it, I think it's better to leave full control of the chunks to the user (i.e., if `chunks=None`, also use `chunks=None` in the `broadcast_to_shape` call instead of the cube's chunks). Otherwise, there is no way a user can specify to use "no special chunking", which might be confusing.

Sorry for all the back and forth, with this I am happy with the PR!

**Review by trexfeathers — APPROVED** [review]
OK these changes make equal sense to me

**Pull request merged**